pkgs/by-name/bo/boxflat/package.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  fetchFromGitHub,
+  gtk4,
+  libadwaita,
+  python3Packages,
+  gobject-introspection,
+  wrapGAppsHook4,
+  copyDesktopItems,
+  makeDesktopItem,
+  nix-update-script,
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "boxflat";
+  version = "1.27.1";
+
+  src = fetchFromGitHub {
+    owner = "Lawstorant";
+    repo = "boxflat";
+    tag = "v${version}";
+    hash = "sha256-W+pnoSyTJS+ORb8vHX4QXOxkDskTb7X2cv6WyU3ctGQ=";
+  };
+
+  format = "setuptools";
+
+  propagatedBuildInputs = [
+    gtk4
+    libadwaita
+
+    python3Packages.pyyaml
+    python3Packages.psutil
+    python3Packages.pyserial
+    python3Packages.pycairo
+    python3Packages.pygobject3
+    python3Packages.evdev
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    wrapGAppsHook4
+    gobject-introspection
+  ];
+
+  preBuild = ''
+    cat > setup.py << EOF
+    import shutil
+    from setuptools import setup
+
+    with open('requirements.txt') as f:
+        install_requires = f.read().splitlines()
+
+    shutil.copyfile('entrypoint.py', 'boxflat/entrypoint.py')
+
+    setup(
+      name='boxflat',
+      packages=['boxflat', 'boxflat.panels', 'boxflat.widgets'],
+      version='${version}',
+      install_requires=install_requires,
+      entry_points={
+        'console_scripts': ['boxflat=boxflat.entrypoint:main']
+      },
+    )
+    EOF
+  '';
+
+  preInstall = ''
+    mkdir -p $out/{usr/share/boxflat,lib/udev/rules.d,share/icons}
+    cp -r data "$out/usr/share/boxflat/"
+    cp -r icons $out/share/icons/hicolor
+    cp udev/99-boxflat.rules "$out/lib/udev/rules.d/"
+  '';
+
+  postInstall = ''
+    wrapProgram "$out/bin/boxflat" \
```

Comment:
We are not double wrapping here, are we?

Reply:
I'm not sure what you mean by double wrapping, I'm only wrapping the program once for the added data-path argument, are you referring to the setup-tools as another wrapper?

Is there another way to do this?

Reply:
```
[nix-shell:~/.cache/nixpkgs-review/pr-375229-2/nixpkgs]$ ls -lah result/bin/
total 80K
dr-xr-xr-x 1 root root  198 Jan  1  1970 .
dr-xr-xr-x 1 root root   50 Jan  1  1970 ..
-r-xr-xr-x 2 root root  20K Jan  1  1970 boxflat
-r-xr-xr-x 2 root root  20K Jan  1  1970 .boxflat-wrapped
-r-xr-xr-x 2 root root  16K Jan  1  1970 .boxflat-wrapped_
-r-xr-xr-x 2 root root  20K Jan  1  1970 ..boxflat-wrapped-wrapped
-r-xr-xr-x 2 root root 1,2K Jan  1  1970 ...boxflat-wrapped-wrapped-wrapped
```
Seems to be tripple(?) wrapped.
Using these three options fixes this.
```nix
    dontWrapGApps = true;
  	makeWrapperArgs = [ ''--add-flags "--data-path $out/usr/share/boxflat/data"'' ];
  	preFixup = ''
    	makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
  	'';

```

Reply:
Yep, it is tripple wrapped which is not great.